File: apps/backend/src/main/java/com/creditn/api/dto/PaymentResponse.java
package com.creditn.api.dto;

import com.creditn.domain.entity.Payment;
import com.creditn.domain.entity.enums.PaymentStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record PaymentResponse(
        Long id,
        Long moimId,
        String merchantName,
        String category,
        BigDecimal amount,
        LocalDateTime approvedAt,
        PaymentStatus status
) {
    public static PaymentResponse from(Payment p) {
        return new PaymentResponse(
                p.getId(),
                p.getMoim().getId(),
                p.getMerchantName(),
                p.getCategory(),
                p.getAmount(),
                p.getApprovedAt(),
                p.getStatus()
        );
    }
}

File: apps/mobile/lib/models/payment_response.dart
class PaymentResponse {
  final int id;
  final int moimId;
  final String merchantName;
  final String? category;
  final double amount;
  final String approvedAt;
  final String status;

  PaymentResponse({
    required this.id,
    required this.moimId,
    required this.merchantName,
    this.category,
    required this.amount,
    required this.approvedAt,
    required this.status,
  });

  factory PaymentResponse.fromJson(Map<String, dynamic> json) {
    return PaymentResponse(
      id: json['id'] as int,
      moimId: (json['moimId'] as int?) ?? 0,
      merchantName: (json['merchantName'] as String?) ?? '',
      category: json['category'] as String?,
      amount: ((json['amount'] ?? 0) as num).toDouble(),
      approvedAt: (json['approvedAt'] as String?) ?? '',
      status: (json['status'] as String?) ?? 'APPROVED',
    );
  }
}
